src/Common/src/Interop/Windows/SChannel/Interop.Sec_Application_Protocols.cs

```diff
@@ -0,0 +1,65 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net.Security;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    [StructLayout(LayoutKind.Sequential, Pack = 1)]
+    internal struct Sec_Application_Protocols
+    {
+        private static readonly int ProtocolListOffset = Marshal.SizeOf<Sec_Application_Protocols>();
+        private static readonly int ProtocolListConstSize = ProtocolListOffset - (int)Marshal.OffsetOf<Sec_Application_Protocols>(nameof(ProtocolExtenstionType));
+        public uint ProtocolListsSize;
+        public ApplicationProtocolNegotiationExt ProtocolExtenstionType;
+        public short ProtocolListSize;
+
+        public static unsafe byte[] ToByteArray(List<SslApplicationProtocol> applicationProtocols)
```

Comment:
@stephentoub, should we also have a method that takes in a span and puts the bytes into it, i.e. TryCopyTo(Span<byte> buffer)?

Reply:
This is internal.  Is the equivalent exposed publicly somewhere?

Reply:
I don't think it's exposed, but: a) maybe we should expose it, b) maybe the internal code that uses it would be written with less copies/allocations if we had such internal API.

Reply:
For b), at that point it's an implementation detail and we should do whatever's most efficient, assuming it matters.

For a), what would it be used for?